sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/LoggingUtils.java

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.logging;
+
+import com.azure.core.util.CoreUtils;
+
+/**
+ * Contains utility methods for logging.
+ */
+public class LoggingUtils {
+    private static final char CL = '\r';
+    private static final char RF = '\n';
+
+    /*
+     * Removes CRLF pattern in the {@code logMessage}.
+     *
+     * @param logMessage The log message to sanitize.
+     * @return The updated logMessage.
+     */
+    public static String sanitizeLogMessageInput(String logMessage) {
+        if (CoreUtils.isNullOrEmpty(logMessage)) {
+            return logMessage;
+        }
+
+        StringBuilder sb = null;
+        int prevStart = 0;
+
+        for (int i = 0; i < logMessage.length(); i++) {
+            if (logMessage.charAt(i) == CL || logMessage.charAt(i) == RF) {
```

Comment:
Was it possible that the regex previously being used over-matched? Since it was `\r` or `\n` but shouldn't it have been `\r\n` or `\n`?

Reply:
I guess whichever combination of special characters in log message is probably added by mistake, there is no harm in removing '\r', '\n' or '\r\n' since log consumers are human beings